### ISSUE TITLE
Visual Studio 2015 (update 2) complains about incomplete type with macros

### ISSUE THREAD

**Issue by Amadeus-** [andrivet/ADVobfuscator#10]
For example, Visual Studio considers the OBFUSCATED4() macro below as "incomplete" -- so, although it will compile just fine, it underlines every instance of it in red and complains non-stop with warnings.
```
#define DEF_OBFUSCATED4(str) MetaString4<andrivet::ADVobfuscator::MetaRandom<__COUNTER__, 3>::value, andrivet::ADVobfuscator::MetaRandomChar4<__COUNTER__>::value, Make_Indexes<sizeof(str) - 1>::type>(str)

#define OBFUSCATED4(str) (DEF_OBFUSCATED4(str).decrypt())
```

Although I'm sure there are many ways to fix it (which is why I'm not doing a pull request), the way I dealt with it was by making it just one macro:
```
#define OBFUSCATED4(str)	(MetaString4<andrivet::ADVobfuscator::MetaRandom<__COUNTER__, 3>::value, andrivet::ADVobfuscator::MetaRandomChar4<__COUNTER__>::value, Make_Indexes<sizeof(str) - 1>::type>(str)).decrypt()
```

Is this a new C++11 standard?   I don't recall this being an issue with previous versions of Visual Studio.

**Comment by andrivet** [comment]
This is a bug in Visual Studio 2015 Intellisense. Sometimes, cleaning the project or just closing/reopening it fix the issue.

**Comment by Amadeus-** [comment]
No, I tried that ..even tried deleting the intellisense DB and recreating it. ...the problem is still there.

There were a lot of new things that came up with the preprocessor with Visual Studio 2015 ...I think they're trying to convince coders to be more explicit with it or something.

But regardless, it's not a big deal for me to change it as I indicated.    I'm actually surprised you wouldn't prefer to have one macro rather than two.
